### ISSUE TITLE
Doc generation broken using json file is broken due to upstream ESM/CJS changes

### ISSUE THREAD

**Issue by hrkeni** [asyncapi/generator#927]
#### Describe the bug
Doc generation broken using json file is broken. This appears to be due to a new upstream release https://github.com/openapi-contrib/openapi-schema-to-json-schema/releases/tag/v3.3.0

#### How to Reproduce
Repo with minimal reproducible bug:

https://github.com/hrkeni/asyncapi-generator-esm-bug

Clone and run `npm run gen-docs`

```
> test@1.0.0 gen-docs
> ag asyncapi.json @asyncapi/html-template --force-write -o ./public

/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/lib/converters/schema.js:8
const lodash_es_1 = require("lodash-es");
                    ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/keni/oss/asyncapi-generator-esm-bug/node_modules/lodash-es/lodash.js from /home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/lib/converters/schema.js not supported.
Instead change the require of lodash.js in /home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/lib/converters/schema.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/lib/converters/schema.js:8:21)
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/lib/convert.js:7:34)
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@openapi-contrib/openapi-schema-to-json-schema/dist/cjs/index.js:8:19)
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@asyncapi/openapi-schema-parser/index.js:1:22)
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@asyncapi/generator/lib/generator.js:10:29)
    at Object.<anonymous> (/home/keni/oss/asyncapi-generator-esm-bug/node_modules/@asyncapi/generator/cli.js:8:19) {
  code: 'ERR_REQUIRE_ESM'
}
```

#### Expected behavior
Dependencies should be resolved correctly

**Comment by jonaslagoni** [comment]
Thanks for the issue!

Seems like we need to freeze v3.2:  https://github.com/openapi-contrib/openapi-schema-to-json-schema/issues/54

It needs to be done in https://github.com/asyncapi/openapi-schema-parser

@hrkeni would you be able to do that?

**Comment by hrkeni** [comment]
@jonaslagoni Happily. Thanks for the suggested fix.

**Comment by iK4tsu** [comment]
we just got hit by this in production as well
shouldn't all dependencies used be locked to a specific version?
the way is ATM, not only could this happen again with any other dependency but it also prevents reproducible builds

**Comment by Gyllsdorff** [comment]
> shouldn't all dependencies used be locked to a specific version?

Yes, but how that is solved depends on if we use `npm ci` and a `package.json`/`package.lock` file or if we do `npx -p @asyncapi/cli`. 

This error scenario is one of the weaknesses of the `npx` command. :) 

I fixed this on my end by using the `npm ci && asyncapi` [pattern](https://www.asyncapi.com/docs/tools/generator/usage#generator-cli) instead.

**Comment by derberg** [comment]
hey folks, how are you install CLI? `npm`? or some other manager?

package-lock.json is locked to 3.1 so 3.3 should not be installed, unless you are using some different tool to install than `npm`. 

@Gyllsdorff you mean `npx` ignores lock file?

**Comment by Gyllsdorff** [comment]
> @Gyllsdorff you mean npx ignores lock file?

`npx` ignores both the lock file, the local npm cache and the globally installed modules. The core value proposition of `npx` (compared to `npm`) is that it can be run in whatever folder and context you want to without relying on additional files.

https://www.npmjs.com/package/clear-npx-cache
https://stackoverflow.com/questions/63510325/how-can-i-clear-the-central-cache-for-npx

**Comment by smoya** [comment]
I think the only way to ensure modules are in sync with `package-lock.json` is `npm ci` https://docs.npmjs.com/cli/v9/commands/npm-ci

**Comment by M3lkior** [comment]
Same issue here ; my pipeline don't succeed anymore

**Comment by derberg** [comment]
Bot closed the issue but there are issues with GitHub Actions 

```
Update - Actions is now experiencing degraded availability. We are still investigating and will provide an update when we have one.
Mar 29, 2023 - 14:43 UTC
Investigating - We are investigating reports of degraded performance for Actions.
Mar 29, 2023 - 14:21 UTC
```

I'll keep you posted when you can actually check if issue is fixed.

**Comment by Gyllsdorff** [comment]
`npx -p @asyncapi/cli asyncapi generate fromTemplate <other stuff>` works for me now.

**Comment by hrkeni** [comment]
I can confirm that my original issue is fixed now

**Comment by M3lkior** [comment]
The problem is that the `package-lock.json` on the generator repo target a `2.0.1` version 
```
  "@asyncapi/openapi-schema-parser": {
      "version": "2.0.1",
      "resolved": "https://registry.npmjs.org/@asyncapi/openapi-schema-parser/-/openapi-schema-parser-2.0.1.tgz",
      "integrity": "sha512-algbtdM1gcAOa8+V8kp7WeBhdaNac82jmZUXx8YjyNfRVo02N2juDrjeBAGJd+FNva9Mb4MM7qfkJoAFpTL5VQ==",
      "requires": {
        "@openapi-contrib/openapi-schema-to-json-schema": "^3.0.0"
      }
    },
```

And the fix is present on a 2.0.2 version

**Comment by derberg** [comment]
@M3lkior yeah, pipelines hiccups. Fix on the way https://github.com/asyncapi/openapi-schema-parser/pull/170
